ember_debug/adapters/web-extension.js

```diff
@@ -77,28 +77,36 @@ export default BasicAdapter.extend({
   }
 });
 
-/**
- * Recursively clones all arrays. Needed because Chrome
- * refuses to clone Ember Arrays when extend prototypes is disabled.
- *
- * If the item passed is an array, a clone of the array is returned.
- * If the item is an object or an array, or array properties/items are cloned.
- *
- * @param {Mixed} item The item to clone
- * @return {Mixed}
- */
-function deepClone(item) {
-  let clone = item;
-  if (isArray(item)) {
-    clone = new Array(item.length);
-    item.forEach((child, key) => {
-      clone[key] = deepClone(child);
-    });
-  } else if (item && typeOf(item) === 'object') {
-    clone = {};
-    keys(item).forEach(key => {
-      clone[key] = deepClone(item[key]);
-    });
-  }
-  return clone;
+let deepClone;
+
+if (Ember.ENV.EXTEND_PROTOTYPES.Array) {
+  deepClone = function deepClone(item) {
```

Comment:
Just wanted to confirm, this is so that we can send the cloned item through the postmessage right? I ask because I want to confirm that we don't actually need to worry about deepClone being used just to be able to mutate the value...

Reply:
Yes. As far as. I can tell this is related to structure cloning. It’s done as one of the last steps before passing to chrome’s postMessage.